src/zenml/hooks/alerter_hooks.py

```diff
@@ -0,0 +1,83 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Functionality for standard hooks."""
+
+from typing import TYPE_CHECKING
+
+from zenml.logger import get_logger
+
+logger = get_logger(__name__)
+
+
+if TYPE_CHECKING:
+    from zenml.steps import BaseParameters, StepContext
+
+
+def alerter_failure_hook(
+    context: "StepContext", params: "BaseParameters", exception: Exception
+) -> None:
+    """Standard failure hook that executes after step fails.
+
+    This hook uses any `BaseAlerter` that is configured within the active stack to post a message.
+
+    Args:
+        context: Context of the step.
+        params: Parameters used in the step.
+        exception: Original exception that lead to step failing.
+    """
+    message = "*Failure Hook Notification! Step failed!*" + "\n\n"
+    message += f"Pipeline name: `{context.pipeline_name}`" + "\n"
+    message += f"Run name: `{context.run_name}`" + "\n"
+    message += f"Step name: `{context.step_name}`" + "\n"
+    message += f"Exception: `({type(exception)}) {exception}`" + "\n\n"
+    message += (
+        f"Step Cache Enabled: `{'True' if context.cache_enabled else 'False'}`"
+        + "\n"
+    )
+
+    if context.stack and context.stack.alerter:
+        context.stack.alerter.post(message)
```

Comment:
This if check should be at the top to even avoid computing the message if no alerter exists.